### ISSUE TITLE
valid-jsdoc rule is too strict

### ISSUE THREAD

**Issue by DarkPark** [eslint/eslint#1598]
here is a simple class jsdoc comments:

```js
/**
 * Base component implementation.
 *
 * @constructor
 * @extends module:stb/emitter~Emitter
 */
```

it is fully recognizable by jsdoc generator but valid-jsdoc rule is telling it's wrong

**Comment by nzakas** [comment]
@Constellation any plans on when this fix will be available? Thanks.

**Comment by alexindigo** [comment]
:+1:

**Comment by nzakas** [comment]
Ping @Constellation are you still planning on working on this?

**Comment by gyandeeps** [comment]
I think this issue will get resolved once https://github.com/eslint/doctrine/pull/102 lands.

**Comment by nzakas** [comment]
Can someone double check this with Doctrine 0.7.0?

**Comment by gyandeeps** [comment]
The fix for this hasn't landed in doctrine yet.

**Comment by nzakas** [comment]
Are you sure? There were some unreleased changes that I just pushed, so it would be best for someone to check to make sure.

**Comment by gyandeeps** [comment]
The above jsdoc doesnt throw any errors

**Comment by nzakas** [comment]
I'm not sure about merging that, it has some problems.

**Comment by knksmith57** [comment]
+1

**Comment by nzakas** [comment]
@knksmith57 thanks. We don't actually do +1s on this repo. This issue has been marked as accepted, which means we are going to do it, and blocked, which means we have an external dependency bug that we are waiting to get fixed.

**Comment by bobholt** [comment]
Closing #4350 as a duplicate of this issue. For additional verification purposes of this fix, the failing code snippet is:

```
/**
 * Creates a new email sending service
 * @class
 * @alias EmailService
 * @requires xhr
 * @requires module:config
 * @requires module:util/auth
 * @requires module:modules/notifications
 */
```

**Comment by nathanmarks** [comment]
@nzakas I was reading through eslint/doctrine#102 and saw [your comment](https://github.com/eslint/doctrine/pull/102#issuecomment-154597170) referencing http://usejsdoc.org/about-namepaths.html and you state " i don't believe `-` is allowed at all.".

I just took a peek at  the jsdoc page linked there, and at the bottom:

```
/**
 * Now we define an event in our {@link chat."#channel"} namespace.
 * @event chat."#channel"."op:announce-motd"
 */
```

> Above is an example of a namespace with "unusual" characters in its member names (the hash character, dashes, even quotes). To refer to these you just need quote the names: chat."#channel", chat."#channel"."op:announce-motd", and so on. Internal quotes in names should be escaped with backslashes: chat."#channel"."say-\"hello\"".

So it appears we should be able to surround the dashed namespace in double quotes.
